### PR TITLE
feat: pass config to NamedTuple fields

### LINKED ISSUE #4219: arbitrary_types_allowed does not propagate to the model which is created for a NamedTuple
### Checks

* [x] I added a descriptive title to this issue
* [x] I have searched (google, github) for similar issues and couldn't find anything
* [x] I have read and followed [the docs](https://pydantic-docs.helpmanual.io/) and still think this is a bug

# Bug

When using a NamedTuple as a field in a pydantic model, it cannot have arbitrary types - even if `arbitrary_types_allowed` is `True` on the model. I found that when `create_model_from_namedtuple` is [called](https://github.com/samuelcolvin/pydantic/blob/master/pydantic/validators.py#L561), `config` is not passed (but for `TypedDict`, it _is_ [passed](https://github.com/samuelcolvin/pydantic/blob/master/pydantic/validators.py#L585)).

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
             pydantic version: 1.9.1
            pydantic compiled: True
                 install path: /home/rory/lab/mres/task/co/.venv/lib/python3.9/site-packages/pydantic
               python version: 3.9.13 (main, May 27 2022, 11:07:19)  [GCC 12.1.0]
                     platform: Linux-5.18.7-arch1-1-x86_64-with-glibc2.35
     optional deps. installed: ['typing-extensions']
```

```py
from typing import NamedTuple
import pydantic

class Foo:
    pass

class Bar(NamedTuple):
    foo: Foo

class MyModel(pydantic.BaseModel):
    bar: Bar

    class Config:
        arbitrary_types_allowed = True
```
image

### PR DESCRIPTION
**Pull request by synek**
## Change Summary

This PR changes the way pydantic validates `NamedTuple` fields. Currently, a new vanilla `BaseModel` is created dynamically for the field, and used for validation. This PR simply passes the parent model's `Config` to that newly created model, allowing settings like `allow_arbitrary_types` to be propagated to the attributes of `NamedTuple` fields.

## Related issue number

fix #4219

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by synek** [comment]
please review

**Review by PrettyWood — APPROVED** [review]
Thanks good catch! Passing the config to TypedDict has been added recently. We should have done it for NamedTuple at the same time

**Comment by synek** [comment]
Sweet, thanks for the quick review :heart:

File: tests/test_annotated_types.py
Comment (hramezani) [inline tests/test_annotated_types.py:167]: I think this line can be removed!

**Comment by hramezani** [comment]
@synek Thanks for your patch :+1: 
Just a small comment. Other than that LGTM!

**Comment by hramezani** [comment]
Please update

**Review by hramezani — APPROVED** [review]

File: tests/test_annotated_types.py
Comment (hramezani) [reply tests/test_annotated_types.py:167]: Ok, let's do it like that. Thanks!

**Comment by synek** [comment]
Thanks for the review @hramezani. Do you need any other changes, or is the PR ready?

**Comment by hramezani** [comment]
> Thanks for the review @hramezani. Do you need any other changes, or is the PR ready?

No, we actually waiting for @samuelcolvin for final review

**Comment by samuelcolvin** [comment]
thanks so much.

**Pull request merged**
